os/kernel/log_dump/log_dump.c

```diff
@@ -169,21 +200,88 @@ int log_dump_save(char ch)
 			current_ptr = 0; /* reset the read pointer as head is modified */
 			node->arr[0] = ch;
 			sq_addlast((sq_entry_t *)node, &log_dump_chunks);
+			if (set_comp_head) { /* new node, so compressed block starts at 0 */
+				node->comp_head = 0;
+				set_comp_head = false;
+			} else {
+				node->comp_head = -1;
+			}
 			log_dump_mem_check(max_size);
 		}
 
 	} else { /* there is place in tail */
-
+		struct log_dump_chunk_s *log_dump_tail = (struct log_dump_chunk_s *)sq_tail(&log_dump_chunks);
 		log_dump_tail->arr[curbytes] = ch;
 		curbytes += 1;
+		if (set_comp_head) { /* new node, so compressed block starts at 0 */
+			if (log_dump_tail->comp_head < 0) { /* compressed node not present */
+				log_dump_tail->comp_head = curbytes-1;
+			}
+                        set_comp_head = false;
+                }
 		if (curbytes % CONFIG_LOG_DUMP_MEMCHECK_SIZE == 0) {
 			log_dump_mem_check(max_size);
+			return LOG_DUMP_OK_FSZ;
 		}
 	}
 
 	return LOG_DUMP_OK;
 }
 
+int log_dump_save(char ch)
+{
+	size_t free_size = 0;
+	int ret = 0;
+
+	uncomp_array[uncomp_curbytes] = ch;
+	uncomp_curbytes++;
+	if (uncomp_curbytes == CONFIG_LOG_DUMP_CHUNK_SIZE) {
+		/* compress the block, add it to the nodes, reset the uncomp_array */
+		set_comp_head = true;
+		unsigned char *out_buf = (unsigned char *)malloc(CONFIG_LOG_DUMP_CHUNK_SIZE + LZMA_PROPS_SIZE);
+		int writesize = CONFIG_LOG_DUMP_CHUNK_SIZE;
+
+		ret = LzmaCompress(&out_buf[LZMA_PROPS_SIZE], &writesize, uncomp_array, CONFIG_LOG_DUMP_CHUNK_SIZE, out_buf, &propsSize, 0, 1<<13 , -1, -1, -1, -1, 1);
+		if (ret != Z_OK) {
+			kmm_free(out_buf);
+			return ret;
```

Comment:
Original `log_dump_save` returns negative value, but in this case, `ret` can be positive even if this is failure case.
Because `LzmaCompress` returns positive value even if fails.
(You can refer to https://github.com/Samsung/TizenRT/blob/master/os/include/tinyara/lzma/7zTypes.h)
Could you check and match the return?

Reply:
done